src/libraries/System.Private.Uri/src/System/IriHelper.cs

```diff
@@ -26,57 +25,33 @@ internal static bool CheckIriUnicodeRange(char unicode, bool isQuery)
         // it checks if the combined char is in the range
         // Takes in isQuery because iri restrictions for query are different
         //
-        internal static bool CheckIriUnicodeRange(char highSurr, char lowSurr, ref bool surrogatePair, bool isQuery)
+        internal static bool CheckIriUnicodeRange(char highSurr, char lowSurr, out bool isSurrogatePair, bool isQuery)
         {
-            bool inRange = false;
-            surrogatePair = false;
-
             Debug.Assert(char.IsHighSurrogate(highSurr));
 
-            if (char.IsSurrogatePair(highSurr, lowSurr))
+            if (isSurrogatePair = char.IsLowSurrogate(lowSurr))
```

Comment:
This looks like a mistake (e.g. `=` instead of `==`) even though it's not.  I realize you probably did it this way in order to avoid an indirect read on isSurrogatePair, but I think for clarity it would be better to do that or use a temp.
```suggestion
            isSurrogatePair = char.IsLowSurrogate(lowSurr);
            if (isSurrogatePair)
```

Reply:
Or if you want to avoid the indirect read, just do:
```C#
if (char.IsLowSurrogate(lowSurr))
{
    isSurrogatePair = true;
    ...;
}

isSurrogatePair = false;
return false;

Reply:
As a bonus, the codegen looks better as well this way

Reply:
> the codegen looks better as well this way

Which way?

Reply:
Both of your suggested ways. [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAZgAJiAmcgFRlwwFgAoAb1fK8oEYlzgECABtyAYQAWMMAGsAklACWAVQB2iyABMYAJWyqA5jB4AKMBOxRyExQYkBlAK5QoacucvlhEAO5OXbhCOGAJCooq4/lAQBtgYMAAK2IquoSLkEQCKjjBQAJ4AlJzcHCzc5RkAZuQmEVExcYnJVgC87hZQAHRyuAAyvvWx8Sbefs5QBUVlFVylMzOOiqohwZWkNG0mi8sFHl1iEKoAbrkYtBDKGGtUJjZ2UW6jUQUA3KzF89zEAOw1W1frchwcgABgQPBBkJBBXIAB5NmCeAAxFEAESBoPBUOh5EIhA+nwq/2uGLBVGxMPhNTJKKR6OBZIpuPx00JRKyOXy5AAZNyCWyZiZiYCGQgkUyqSYwbS6aSxUy8fyBeVhTRRRCJQisbT6ZiNVDJpM3qyKgBfJVcH7kSrYYS4GDG8rmk3EPhpUSSaTyJRqDQQbR6QwwG57ay2BzjNyhp6R8hBEKCdJ1cYNeJJFJuRPhXDZXKFAlzebJlyppopchtPbdPoDFNDGAjWsuV4Wqo1YvRevpiatwts7YrAEbGoD3YdToHY6nc6Xa63cMPLxNibG1vlK1C1YivWa6ngnVy/U4xUm5WquXkg1wrVUA+iy+QmEn5Uqjl5nl808vribody8VXpK0p3piAGPsya4Cue6rYjiQHamih6wYaLZflwzqfFaNp2g6BIYV8bpZuIUiyAoKjqFouj6EYpBmB0Yb3LG0bLoEwTuhkkR1o03aZmEHG5vkUzlH2FSKNUdGWNW/RjCW9aNjJExCfMImfB2pbdhW5AYFAOSOgKA7kFuw5bEsGBjpJk4nFAZwXEO86MQES4KahkGWr8v4kjBgFasiiFeeBz7fiOf73ruUoILefmYg+x4skF7Y5pyeQfq5hIedu0phcBUWZVegXxT+RlIVlCGyv5j6GnpMzOq2alds0mnYfaVUVFhtrNXhrCmkAA=).

`if (isSurrogatePair = char.IsLowSurrogate(lowSurr))` looks the worst out of the three.

Reply:
Ah. Yes :wink: